packages/aws-cdk/bin/cdk.ts

```diff
@@ -521,14 +539,19 @@ function yargsNegativeAlias<T extends { [x in S | L ]: boolean | undefined }, S
   };
 }
 
+let SHOW_ALL_NOTICES = false;
 initCommandLine()
-  .then(value => {
-    if (value == null) { return; }
+  .then(async ({ value, configuration }) => {
```

Comment:
moved the `displayNotices()` call here primarily due to the use case of `cdk synth` with one stack. I think we want the notices to display below the template that gets printed to stdout, and that happens here in the `.then()` block.